### PR TITLE
new rule: prefer cap type names

### PR DESCRIPTION
**Pull request by DianaSuvorova**
as per suggestion from #61

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -210,6 +210,34 @@ export const rules = {
       );
     },
   },
+  'type-names-cap': {
```
Comment (jnwng) [inline src/index.js]: @DianaSuvorova this rule name could be more descriptive, especially for folks who don't know what `cap` is short for. what would you think about `capitalized-type-name`? (inspired by [capitalized-comments](http://eslint.org/docs/rules/capitalized-comments) from eslint)

File: src/rules.js [hunk 2]
```diff
@@ -45,3 +45,16 @@ export function RequiredFields(context, options) {
     },
   };
 }
+
+export function typeNamesShouldBeCapitalized(context) {
+  return {
+    NamedType(node) {
```
Comment (jnwng) [inline src/rules.js]: does `NamedType` get called for inline fragments? i.e., can we add a test case for
```graphql
query {
  someUnion {
    ... on someUnionMember {
      someField
    }
  }
}
```

**Comment by jnwng** [comment]
thank you @DianaSuvorova! everything looks good, just a couple of comments to make sure the API is a little more straightforward and captures some edge cases

File: src/rules.js (hunk 2 above)
Comment (DianaSuvorova) [inline src/rules.js]: Added a test.

**Comment by DianaSuvorova** [comment]
@jnwng , thanks for review! I believe I addressed all the comments. Ready for re-review 🙂

**Comment by jnwng** [comment]
awesome, looks good. last comment — does this actually need a schema? we're not verifying it against the schema, right?

**Comment by DianaSuvorova** [comment]
@jnwng , yeah I actually was thinking the same and tried the rule without providing a schema. Looks like ast parser needs it to figure out which node  is `NamedType`. So yes schema is required for the rule.

**Comment by jnwng** [comment]
thank you @DianaSuvorova!
